### PR TITLE
ARGO-128 Created utility function and fixes

### PR DESCRIPTION
**Pull request by makistsantekidis**
* Added the functions to authenticate clients to tenants and return their database configuration
* Changed all the uses of mongo.OpenSession to use cfg.MongoDB

### CODE REVIEW

**Comment by makistsantekidis** [comment]
please review

/cc @skanct @pkoro @kaggis

File: utils/authentication/authenticate_test.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+}
+
+func (suite *AuthenticationProfileTestSuite) TestAuthentication() {
+
+	request, _ := http.NewRequest("GET", "", strings.NewReader(""))
+	request.Header.Set("x-api-key", suite.clientkey)
+
+	tenantdbconfig, err := AuthenticateTenant(request.Header, suite.cfg)
+	// Check that we must have a 200 ok code
+	suite.Equal(nil, err, "authetication probelm")
```
Comment (pkoro) [inline utils/authentication/authenticate_test.go]: typo in error message

**Comment by pkoro** [comment]
+1

**Comment by pkoro** [comment]
tests are passing. merging now.

**Pull request merged**